Modules/_testcapi/object.c

```diff
@@ -485,6 +485,41 @@ is_uniquely_referenced(PyObject *self, PyObject *op)
 }
 
 
+static PyObject *
+object_setattr_null_exc(PyObject *self, PyObject *args)
+{
+    PyObject *obj, *name;
+    if (!PyArg_ParseTuple(args, "OO", &obj, &name)) {
+        return NULL;
+    }
+
+    PyErr_SetString(PyExc_ValueError, "error");
```

Comment:
You can pass an exception as argument. Then you can check `__context__` or `__cause__` of SystemError by identity.